src/main/java/org/gridsuite/directory/server/DirectoryService.java

```diff
@@ -0,0 +1,87 @@
+
+    DirectoryService(DirectoryRootRepository directoryRootRepository, DirectoryElementRepository directoryElementRepository) {
+        this.directoryRootRepository = directoryRootRepository;
+        this.directoryElementRepository = directoryElementRepository;
+    }
+
+    private static ElementAttributes fromEntity(DirectoryElementEntity entity) {
+        return new ElementAttributes(entity.getId(), entity.getName(), ElementType.valueOf(entity.getType()), new AccessRightsAttributes(entity.isPrivate()), entity.getOwner());
+    }
+
+    public Mono<DirectoryRootEntity> getRootDirectory() {
+        return directoryRootRepository.findAll().single();
+    }
+
+    public Mono<DirectoryElementEntity> createDirectory(CreateDirectoryAttributes createDirectoryAttributes) {
+        Mono<DirectoryElementEntity> createdDirectory = directoryElementRepository.save(new DirectoryElementEntity(null, createDirectoryAttributes.getParentId(),
+                                                                   createDirectoryAttributes.getDirectoryName(),
+                                                                   ElementType.DIRECTORY.toString(),
+                                                                   createDirectoryAttributes.getAccessRights() != null ? createDirectoryAttributes.getAccessRights().isPrivate() : true,
+                                                                   createDirectoryAttributes.getOwner(),
+                                                        true));
+
+        return createdDirectory;
+    }
+
+    public Mono<DirectoryElementEntity> addElementToDirectory(String directoryUuid, ElementAttributes elementAttributes) {
+        return directoryElementRepository.save(new DirectoryElementEntity(elementAttributes.getElementUuid(),
+                                                                   directoryUuid != null ? UUID.fromString(directoryUuid) : null,
+                                                                   elementAttributes.getElementName(),
+                                                                   elementAttributes.getType().toString(),
+                                                                   elementAttributes.getAccessRights().isPrivate(),
+                                                                   elementAttributes.getOwner(),
+                                                        true));
+    }
+
+    public Flux<ElementAttributes> listDirectoryContent(String directoryUuid) {
+        return directoryElementRepository.findByParentId(directoryUuid != null ? UUID.fromString(directoryUuid) : null).map(DirectoryService::fromEntity);
+    }
+
+    public Mono<Void> renameElement(String elementUuid, String newElementName) {
+        return directoryElementRepository.updateElementName(UUID.fromString(elementUuid), newElementName);
+    }
+
+    public void setDirectoryAccessRights(String directoryUuid, AccessRightsAttributes accessRightsAttributes) {
```

Comment:
! To be implemented ?

Reply:
A minimalist implementation is now provided.